src/Microsoft.AspNetCore.Razor/CodeGenerators/Visitors/CSharpTagHelperFieldDeclarationVisitor.cs

```diff
@@ -151,6 +156,21 @@ private void PreAllocateUnboundTagHelperAttributes(TagHelperChunk chunk)
                                     .WriteStringLiteral(plainText)
                                     .WriteEndMethodInvocation(endLine: false)
                                     .WriteEndMethodInvocation();
+                                }
+                                else
+                                {
+                                    Writer
```

Comment:
Only difference between this set of writes and the above is the value being surrounded with `Html.Raw`/`HtmlString` suggest unifying the pre and post logic and just having an if unbound for the value..